apply changes to .c files to improve compatibility with c++ compilers

fwph opened the pull request:
# Description

The code changes in this PR were generated automatically by the Permanence AI Coder and reviewed by myself. This code improvement PR updates four C source files to improve sections of code which are not compatible with C++. In particular, the fixes address issues with with struct creation (field order mismatch), initialization of char * strings with literals, and unnecessary conversions between integers and enums. 

on-behalf-of: @permanence-ai [github-ai@permanence.ai](mailto:github-ai@permanence.ai)

# Checklist

- No new tests as this is code improvement update without feature changes
- Software path test results had only expected failures (per latest release notes) following the changes:

```
[----------] Global test environment tear-down
[==========] 385 tests from 125 test suites ran. (447894 ms total)
[  PASSED  ] 334 tests.
[  SKIPPED ] 45 tests, listed below:
. . .
[  FAILED  ] 6 tests, listed below:
[  FAILED  ] ta_c_api_deflate_stateful.dynamic_default_verify
[  FAILED  ] ta_c_api_deflate_stateful.dynamic_high_verify
[  FAILED  ] ta_c_api_deflate_stateful.fixed_default_verify
[  FAILED  ] ta_c_api_deflate_stateful.fixed_high_verify
[  FAILED  ] ta_c_api_deflate_stateful.static_default_verify
[  FAILED  ] ta_c_api_deflate_stateful.static_high_verify
```

mzhukova commented on sources/core-sw/src/filtering/qplc_unpack_prle.c:
```diff
@@ -123,7 +123,7 @@ OWN_QPLC_FUN(qplc_status_t, qplc_unpack_prle_8u, (uint8_t * *pp_src,
     uint8_t  *src_ptr      = (uint8_t *) *pp_src;
     uint8_t  *src_stop_ptr = src_ptr + src_length;
     uint8_t  *dst_stop_ptr = dst_ptr + dst_length;
-    uint32_t status        = QPLC_STS_OK;
+    qplc_status_t status        = QPLC_STS_OK;
```
Align the `=` to keep previous formatting?

fwph replied:
that brings up a question we had: is there a clang_format or indent configuration that is preferred for this repository?

mzhukova replied:
good question! clang_format is currently in the works, so for now for any changes we're just trying to be consistent with the code around

fwph replied:
whitespace is the bot’s nemesis so i will push a manual update to this bit of formatting. will be up by Thursday morning.

fwph replied:
updated now

mzhukova commented on sources/core-iaa/sources/driver_loader/hw_configuration_driver.c:
```diff
@@ -191,7 +191,7 @@ bool own_load_configuration_functions(void *driver_instance_ptr) {
         if (err_message) {
             // @todo this is a workaround to optionally load accfg_device_get_iaa_cap
-            char *iaa_cap_func_name = "accfg_device_get_iaa_cap";
+            const char *iaa_cap_func_name = "accfg_device_get_iaa_cap";
```
I'm curious as to why this line was changed, but not the L197 for instance.

fwph replied:
That's a fair question! At the moment, the AI coder is not very good at prioritizing fix attempts; I believe this is a direct consequence of that. I'll try to confirm that as we are actively working on improvements to the selection part of the process now.

mzhukova replied:
sure, I'll set it to pending state meanwhile.

fwph replied:
hi @mzhukova , this branch has been regenerated and pushed -- catching the below change and some additional other cases.

mzhukova commented:
@fwph could you please elaborate with what compiler you faced issues? and what kind of issue was it? a warning of some kind perhaps? 
We have several compilers in testing, and currently do not see any problems, so just wondering how we could improve to catch those early.

fwph commented:
@mzhukova  The approach we used for this PR was to identify updates to the C code that reduce the overhead for potential C++ conversion in the future; these changes are not all that would be necessary if the updated files were converted to C++, but they are changes that are compatible with both C and C++. 

The compiler used to localize the specific cases is g++. The three types of changes are:

- Designator order mismatch, which is an error in g++. This will not compile under C++.
- Enum <-> integer conversion, which is an error in g++. The status variables in these cases should use the enum type.
- `char *` initialization with string literal, which is a warning in g++. This is an ISO C++ violation. With this kind of initialization it's more correct to be a `const char *` rather than a `char *`

C is more lenient for each of these cases, but the updates do represent a greater degree of correctness.

Permanence-AI-Coder committed:
fix designator order mismatch error for 2 files

    - modify sources/c_api/compression_operations/qpl_deflate_slow.c (fix designator order mismatch error)
    - modify sources/isal/igzip/huff_codes.c (fix designator order mismatch error)

on-behalf-of: @permanence-ai <github-ai@permanence.ai>

Permanence-AI-Coder committed:
fix non const char * initialized from string literal for hw_configuration_driver.c

on-behalf-of: @permanence-ai <github-ai@permanence.ai>

Permanence-AI-Coder committed:
fix type conversion error for qplc_unpack_prle.c

on-behalf-of: @permanence-ai <github-ai@permanence.ai>

fwph commented:
branch has been regenerated to fix conflicts and apply new clang-format standards

mzhukova commented:
These changes were merged with f55c1d471c95419aeb7e2dc329a2876705bb99a4, thanks for the contribution!
